hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -68,10 +67,10 @@ class ServiceBackend() extends Backend {
     assert(previous == null)
   }
 
-  def userContext[T](username: String, timer: ExecutionTimer)(f: (ExecuteContext) => T): T = {
+  def userContext[T](username: String, sessionID: String, timer: ExecutionTimer)(f: (ExecuteContext) => T): T = {
     val user = users.get(username)
     assert(user != null, username)
-    ExecuteContext.scoped(user.tmpdir, "file:///tmp", this, user.fs, timer, null)(f)
+    ExecuteContext.scoped(user.tmpdir, "file:///tmp", this, user.fs.asCacheable(sessionID), timer, null)(f)
```

Comment:
Hmm. Tim has a PR that's going to make the user's code use the fs in the execute context. I think this has to be a normal FS until we can properly stream into Memory.